src/core/scheduler.rs

```diff
@@ -29,20 +29,27 @@ impl CancelHandle {
 
 /// Schedule a task to run periodically.
 /// Starts a new thread for every task.
+///
+/// # Panics
+///
+/// Panics if the OS fails to create a thread.
 fn set_schedule<F>(every: Duration, operation: F) -> CancelHandle
 where
     F: Fn() -> () + Send + 'static,
 {
     let handle = CancelHandle::new();
     let inner_handle = handle.clone();
 
-    thread::spawn(move || loop {
-        thread::sleep(every);
-        if inner_handle.is_cancelled() {
-            break;
-        }
-        operation();
-    });
+    thread::Builder::new()
+        .name("dipstick-scheduler".to_string())
+        .spawn(move || loop {
+            thread::sleep(every);
+            if inner_handle.is_cancelled() {
+                break;
+            }
+            operation();
+        })
+        .unwrap(); // TODO: Panic, change API to return Result?
```

Comment:
I would `log!` the error so it doesn't get lost. There could be a optional error-handler so you can restart the job. Schedules are not supposed to return a value.